### PR TITLE
Cart checkout zth, Completes User 25 - 26

### PR DESCRIPTION
**Pull request by zachholcomb**
Completes User story 25 and 26
Adds user to order one to many relationship
Fixes bug in user register being redirected to profile
Fixes password digest bug in user register error message
Adds functionality that cart cannot checkout requires a user to be logged in to continue
Updates old tests to create orders with a user
Notes: Need to refactor user controller to use helper methods, need to refactor orders controller to use helper methods
Passing all tests

### CODE REVIEW

File: db/migrate/20200411180309_add_users_to_orders.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddUsersToOrders < ActiveRecord::Migration[5.1]
+  def change
```
Comment (zachholcomb) [inline db/migrate/20200411180309_add_users_to_orders.rb:2]: had to create user order one to many relationship here to be able to get a users orders

File: db/schema.rb [hunk 2]
```diff
@@ -66,6 +66,9 @@
     t.integer "zip"
     t.datetime "created_at", null: false
     t.datetime "updated_at", null: false
+    t.bigint "user_id"
+    t.string "status", default: "Pending"
```
Comment (zachholcomb) [inline db/schema.rb:70]: Also gave new orders a default of Pending

File: spec/features/cart/checkout_spec.rb [hunk 3]
```diff
@@ -19,6 +19,18 @@
     end
 
     it 'Theres a link to checkout' do
+      user = User.create!(name: "Steve",
+                          address:"123 Main St.",
```
Comment (zachholcomb) [inline spec/features/cart/checkout_spec.rb:23]: in a lot of these old tests they did not include a user to create orders/checkout

File: config/routes.rb [hunk 4]
```diff
@@ -53,6 +53,8 @@
   get '/profile/edit/password', to: 'users#edit_password'
   get "/profile/edit", to: "users#edit"
   patch "/profile", to: "users#update"
+  get "/profile/orders", to: "orders#index"
```
Comment (zachholcomb) [inline config/routes.rb:56]: i think this is probably a good use case to use nested route resources

File: app/models/user.rb [hunk 5]
```diff
@@ -2,8 +2,10 @@ class User < ApplicationRecord
   validates_presence_of :name, :address, :city, :state, :zip
   validates :email, uniqueness: true, presence: true
   validates_presence_of :password_digest, require: true
+  after_validation { self.errors.messages.delete(:password_digest) }
```
Comment (steve-meyers) [inline app/models/user.rb:5]: I like this solution!

File: app/views/cart/show.html.erb [hunk 6]
```diff
@@ -1,5 +1,7 @@
 <h1>Cart</h1>
-
+<% if !current_user %>
+  <p>To checkout you must <%= link_to 'log in', '/login' %> or <%= link_to 'register', '/register' %></p>
```
Comment (zachholcomb) [inline app/views/cart/show.html.erb:3]: this message will persist to encourage the visitor to log in or register before checking out

File: app/models/user.rb (hunk 5 above)
Comment (zachholcomb) [inline app/models/user.rb:5]: this line of code deletes password password digest from error messages

**Review by jrsewell400 — APPROVED** [review]
Everything looks good to me!

File: app/controllers/orders_controller.rb [hunk 7]
```diff
@@ -1,25 +1,34 @@
 class OrdersController <ApplicationController
+  def index
+    user = User.find(session[:user])
+    @orders = user.orders
+  end
 
   def new
-
+    if !current_user
```
Comment (zachholcomb) [inline app/controllers/orders_controller.rb:8]: This is a sad path for visitors who click checkout without being logged in or registered

File: spec/features/cart/checkout_spec.rb [hunk 8]
```diff
@@ -36,4 +48,107 @@
...
+# When I add items to my cart
+# And I visit my cart
+# I see a button or link indicating that I can check out
+# And I click the button or link to check out and fill out order info and click create order
+# An order is created in the system, which has a status of "pending"
+# That order is associated with my user
+# I am taken to my orders page ("/profile/orders")
+# I see a flash message telling me my order was created
+# I see my new order listed on my profile orders page
+# My cart is now empty
```
Comment (steve-meyers) [inline spec/features/cart/checkout_spec.rb:154]: We need to remove these commented out lines.

File: spec/features/cart/checkout_spec.rb (hunk 3 above)
Comment (elguapogordo) [reply spec/features/cart/checkout_spec.rb:23]: same with my user(role: 1) were created in tests without a merchant. dashboard was breaking those. fixed now.

File: app/controllers/users_controller.rb [hunk 9]
```diff
@@ -6,6 +6,7 @@ def new
   def create
     @user = User.new(user_params)
     if @user.save
+      session[:user] = @user.id
```
Comment (jrsewell400) [inline app/controllers/users_controller.rb:9]: good catch and thanks for fixing this error!

File: app/views/cart/show.html.erb (hunk 6 above)
Comment (elguapogordo) [reply app/views/cart/show.html.erb:3]: i like this feature

**Review by elguapogordo — APPROVED** [review]

File: app/controllers/orders_controller.rb [hunk 10]
```diff
@@ -1,25 +1,34 @@
 class OrdersController <ApplicationController
+  def index
+    user = User.find(session[:user])
```
Comment (steve-meyers) [inline app/controllers/orders_controller.rb:3]: Can we just call the current_user method here?

**Review by steve-meyers — APPROVED** [review]
I don't see any major issues.

**Pull request merged**